Asterix/optics/coronagraph.py

```diff
@@ -578,3 +595,182 @@ def HLC(self):
+        Array holding the phase mask, in radians.
+    """
+    xx, yy = np.meshgrid(np.arange(dim) - dim / 2,
+                         np.arange(dim) - dim / 2)
+
+    fqpm_thick_vert = np.zeros((dim, dim))
+    fqpm_thick_vert[np.where(xx < 0)] = 1
+    fqpm_thick_hor = np.zeros((dim, dim))
+    fqpm_thick_hor[np.where(yy >= 0)] = 1
+    fqpm_thick = fqpm_thick_vert - fqpm_thick_hor
+
+    return fqpm_thick
+
+
+def create_wrapped_vortex_mask(dim, thval, phval, jump, return_1d=False, piperiodic=True, offset=0, cen_shift=(0,0)):
+    """
+    Create a wrapped vortex phase mask.
+
+    Analytical calculation of this phase mask coronagraph see [Galicher2020]_.
+
+    .. [Galicher2020] Galicher et al. 2020, "A family of phase masks for broadband coronagraphy example of the wrapped
+            vortex phase mask theory and laboratory demonstration "
+
+    Parameters
+    ----------
+    dim : int
+       Number of pixels for the resulting phase mask.
+    thval : array
+        Angle values at location of phase jumps. First is 0, last is pi, completing a half of a unit circle.
+    phval : array
+        Phase values corresponding to the angle values thval just after the phase jump happened.
+    jump : array
+        Phase jump between phase segments on the unit circle. Has one less element than thval and phval.
+    return_1d : bool
+        If True, return a 1D phase profile, otherwise a 2D phase mask; default False.
+    piperiodic: bool
+        If True, assume max(thval) is pi and return a pi-periodic mask. Needs to be True to get full 2D phase mask.
+    offset : float
+        General offset to the whole ramp; default 0.
+    cen_shift : tuple of floats
+        x- and y-shift of the center of the mask with respect to the center of the array; default (0,0).
+
+    Returns
+    -------
+    angles : array
+        Array holding angle values in radians. 1D array or 2D array depending on return_1d.
+    phase_mask : array
+        Array holding the phase mask, in radians. 1D array or 2D array depending on return_1d.
+
+    Examples
+    --------
+    ### Galicher et al 2020 mask (orientation for THD2 on sep-2022)
+    # Input parameters
+    thval = np.array([0, 3, 4, 5, 8]) * np.pi / 8
+    phval = np.array([3, 0, 1, 2, 1]) * np.pi
+    jump = np.array([2, 2, 2, 2]) * np.pi
+
+    # Create an dplot focal plane mask in 1D
+    angles_1d, phase_1d = create_wrapped_vortex_mask(dim=128, thval=thval, phval=phval, jump=jump, return_1d=True)
+    plt.figure(figsize=(16,8))
+    plt.plot(angles_1d, phase_1d)
+    plt.xlabel("Angle (rad)")
+    plt.ylabel("Phase (rad)")
+    plt.show()
+
+    # Create an dplot focal plane mask in 2D
+    angles_2d, phase_2d = create_wrapped_vortex_mask(dim=128, thval=thval, phval=phval, jump=jump, return_1d=False)
+    plt.figure(figsize=(7,7))
+    plt.imshow(phase_2d, cmap="Reds", origin="lower")
+    plt.colorbar(label="Phase (rad)")
+    plt.show()
+    """
+    if phval.shape != thval.shape:
```

Comment:
I've never coded a WV maybe ask @rgalicher to review this part ?

Reply:
The result is correct, so that might be enough?